.github/main.go

```diff
@@ -156,6 +160,67 @@ func (ci *CI) withSDKWorkflows(runner *dagger.Gha, name string, sdks ...string)
 	return ci
 }
 
+func (ci *CI) withTestWorkflows(runner *dagger.Gha, name string) *CI {
+	w := runner.
+		Workflow(name).
+		WithJob(runner.Job("engine-lint", "engine lint", dagger.GhaJobOpts{
+			Runner: []string{GoldRunner(false)},
+		})).
+		WithJob(runner.Job("scripts-lint", "scripts lint")).
+		WithJob(runner.Job("cli-test-publish", "cli test-publish")).
+		WithJob(runner.Job("engine-test-publish", "engine publish --image=dagger-engine.dev --tag=main --dry-run")).
+		WithJob(runner.Job("scan-engine", "engine scan")).
+		With(splitTests(runner, "test-", false, []testSplit{
+			{"modules", []string{"TestModule"}, GoldRunner(false)},
+			{"module-runtimes", []string{"TestGo", "TestPython", "TestTypescript", "TestElixir", "TestPHP"}, GoldRunner(false)},
+			{"cli-engine", []string{"TestCLI", "TestEngine"}, GoldRunner(false)},
+			// HACK: our main runners don't support cgroupsv2
+			{"cgroupsv2", []string{"TestProvision", "TestTelemetry"}, "ubuntu-latest"},
```

Comment:
these will need memory_overcommit enabled to pass, as discussed in DMs.

Reply:
Added this, bit of a plumbing pain, but should now be done :saluting_face: